examples/anthos-service-mesh-multicluster/infrastructure/shared.tf

```diff
@@ -0,0 +1,180 @@
+  cluster3_master_authorized_networks_config_1_display_name = "all"
+  cluster3_master_authorized_networks_config_1_cidr_block   = "0.0.0.0/0"
+
+  # cluster3 cluster details
+  cluster3_cluster_name                 = "cluster3"
+  cluster3_node_pool_initial_node_count = 1
+  cluster3_cluster_release_channel      = "REGULAR"
+
+  cluster3_node_pool_autoscaling_min_node_count = 1
+  cluster3_node_pool_autoscaling_max_node_count = 5
+
+  cluster3_node_pool_machine_type = "e2-standard-4"
+  cluster3_network_tag            = "cluster3"
+
+  cluster3_node_pool_oauth_scopes = [
+    "https://www.googleapis.com/auth/compute",
+    "https://www.googleapis.com/auth/devstorage.read_only",
+    "https://www.googleapis.com/auth/logging.write",
+    "https://www.googleapis.com/auth/monitoring",
+    "https://www.googleapis.com/auth/trace.append",
+    "https://www.googleapis.com/auth/cloud_debugger",
+    "https://www.googleapis.com/auth/cloud-platform",
+    "https://www.googleapis.com/auth/servicecontrol",
+    "https://www.googleapis.com/auth/service.management.readonly"
+  ]
+  cluster3_node_pool_auto_repair     = true
+  cluster3_node_pool_auto_upgrade    = true
+  cluster3_node_pool_max_surge       = 1
+  cluster3_node_pool_max_unavailable = 0
+
+  # cluster4 network details
+  cluster4_subnet_name = "cluster4"
+  cluster4_subnet_cidr = "10.184.20.0/24"
+
+  cluster4_pod_ip_range_name = "cluster4-pod-cidr"
+  cluster4_pod_ip_cidr_range = "10.185.192.0/18"
+
+  cluster4_services_ip_range_name = "cluster4-services-cidr"
+  cluster4_services_ip_cidr_range = "172.16.3.0/24"
+
+  cluster4_master_ipv4_cidr_block                           = "192.168.3.0/28"
+  cluster4_master_authorized_networks_config_1_display_name = "all"
+  cluster4_master_authorized_networks_config_1_cidr_block   = "0.0.0.0/0"
+
+  # cluster4 cluster details
+  cluster4_cluster_name                 = "cluster4"
+  cluster4_node_pool_initial_node_count = 1
+  cluster4_cluster_release_channel      = "REGULAR"
+
+  cluster4_node_pool_autoscaling_min_node_count = 1
+  cluster4_node_pool_autoscaling_max_node_count = 5
+  cluster4_node_pool_machine_type               = "e2-standard-4"
+  cluster4_network_tag                          = "cluster4"
+
+  cluster4_node_pool_oauth_scopes = [
+    "https://www.googleapis.com/auth/compute",
+    "https://www.googleapis.com/auth/devstorage.read_only",
+    "https://www.googleapis.com/auth/logging.write",
+    "https://www.googleapis.com/auth/monitoring",
+    "https://www.googleapis.com/auth/trace.append",
+    "https://www.googleapis.com/auth/cloud_debugger",
+    "https://www.googleapis.com/auth/cloud-platform",
+    "https://www.googleapis.com/auth/servicecontrol",
+    "https://www.googleapis.com/auth/service.management.readonly"
+  ]
+  cluster4_node_pool_auto_repair     = true
+  cluster4_node_pool_auto_upgrade    = true
+  cluster4_node_pool_max_surge       = 1
+  cluster4_node_pool_max_unavailable = 0
```

Comment:
alot of these can be hardcoded directly within the resource declaration. Using locals for each input adds a layer of indirection and since this is not a module, IMHO it is okay to be directly put into the resource.